src/main/java/org/assertj/assertions/generator/BaseAssertionGenerator.java

```diff
@@ -319,14 +322,54 @@ private String generateAssertionsEntryPointClassContent(final Set<ClassDescripti
         ? determineBestEntryPointsAssertionsClassPackage(classDescriptionSet)
         : entryPointClassPackage;
     entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, PACKAGE, classPackage);
-
+    String imports = buildImports(classDescriptionSet);
+    entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, IMPORTS, imports);
     String allEntryPointsAssertionContent = generateAssertionEntryPointMethodsFor(classDescriptionSet,
                                                                                   entryPointAssertionMethodTemplate);
     entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, ALL_ASSERTIONS_ENTRY_POINTS,
                                                allEntryPointsAssertionContent);
     return entryPointAssertionsClassContent;
   }
 
+  private static String buildImports(Set<ClassDescription> classDescriptionSet) {
+    StringBuilder builder = new StringBuilder();
+    for (String anImport : extractImports(classDescriptionSet)) {
+      builder.append(format(IMPORT_LINE, anImport, LINE_SEPARATOR));
+    }
+    return builder.toString();
+  }
+
+  /**
+   * The set will contain all the FQN of the imports for which simple class names can be used.
+   * <p>
+   * Example:
+   * <pre>
+   *   org.example.Outer -> part of the set
+   *   org.example.Outer.Inner -> org.example.Outer in the result
+   *   org.example.other.Outer -> not in the set because another FQN for the same simple class name Outer already exists
+   *   org.example.other.Outer.Inner -> same as above, another FQN for the same outer class name exists
+   * </pre>
+   *
+   * @param classDescriptionSet for which we need to extract the imports
+   * @return a sorted set with all the FQN that can be imported
+   */
+  private static SortedSet<String> extractImports(Set<ClassDescription> classDescriptionSet) {
+    Map<String, String> importedQualifiedName = new HashMap<>(classDescriptionSet.size() * 2);
+    for (final ClassDescription description : new TreeSet<>(classDescriptionSet)) {
```

Comment:
Not necessarily, do we want to have some ordering of the imports, i.e. Do we want to first methods to always have a normal import and then the methods below to use FQN in case that is needed? What we can do is to pass a `SortedSet<ClassDescription>`  to `generateAssertionsEntryPointClassContent` as we in any case create those down the line.